spieleplattform-frontend/src/pages/Game/AllGamesPage/AllGamesPage.tsx

```diff
@@ -1,40 +1,75 @@
 import Game from "../Game";
+import GameConsole from "../GameConsole";
 import {useEffect, useState} from "react";
 import axios from "axios";
 import GameComponent from "../GameComponent/GameComponent";
-import "./AllGamesPage.css"
+import "./AllGamesPage.scss"
 import Error404Page from "../../Error/Error404Page";
 import ErrorPage from "../../Error/ErrorPage";
+import {useSearchParams} from "react-router-dom";
 
 export default function AllGamesPage() {
-    const [games, setGames] = useState<Game[] | null>(null);
-    const [errorStatus, setErrorStatus] = useState(null);
+    const [games, setGames] = useState<Game[] | null>(null)
+    const [gameConsoles, setGameConsoles] = useState<GameConsole[] | null>(null)
+    const [errorStatus, setErrorStatus] = useState(null)
+
+    const [searchParams, setSearchParams] = useSearchParams()
+    const gameConsoleIdValue = Number(searchParams.get('gameConsoleId'))
+    const gameConsoleId = isNaN(gameConsoleIdValue) ? 0 : gameConsoleIdValue
 
     useEffect(() => {
-        axios.get("/game")
+        if (gameConsoleId === 0)
+            setSearchParams({})
+        const gamesApiPath = "/games" + ((gameConsoleId !== 0) ? "?gameConsoleId=" + gameConsoleId : "")
+
+        axios.get("/gameConsoles")
+            .then((response) => {
+                setGameConsoles(response.data)
+            })
+            .catch(error => {
+                setErrorStatus(error.response.status)
+            })
+
+        axios.get(gamesApiPath)
             .then((response) => {
                 setGames(response.data)
             })
             .catch(error => {
-                setErrorStatus(error.response.status);
+                setErrorStatus(error.response.status)
             })
-    }, [])
+    }, [gameConsoleId, setSearchParams])
 
     if (errorStatus)
         return errorStatus === 404 ? <Error404Page/> : <ErrorPage/>
-    return games && (
-        <div className="flex-container">
-            {games.map(game =>
-                <GameComponent
-                    key={game.id}
-                    id={game.id}
-                    name={game.name}
-                    releaseDate={game.releaseDate}
-                    developer={game.developer}
-                    imgUrl={game.imgUrl}
-                    ratings={game.ratings}
-                />
-            )}
-        </div>
+    return games && gameConsoles && (
+        <>
+            <form className="filter-form">
+                <select name="gameConsoles" id="gameConsoles" value={gameConsoleId} onChange={e => {
+                    const value = e.target.value
+                    if (value === "0")
+                        setSearchParams({})
+                    else
+                        setSearchParams({'gameConsoleId': value})
```

Comment:
Wenn dein onChange Methode nicht wirklich kurz ist, solltest du sie in eine eigene Methode auslagern und nicht innerhalb des jsx schreiben, für mehr Übersichtlichkeit. Davon abgesehen, hab ich das Gefühl, du machst dir die Erstellung des Query strings hier viel komplizierter als sie sein müsste. Axios nimmt ein Objekt entgegen mit der property `params`. Ich würde vermuten, dass du hier einfach deine `searchParams` reingeben kannst. Hier ein Beispiel: https://masteringjs.io/tutorials/axios/get-query-params

Reply:
Oh ja stimmt